src/transport.rs

```diff
@@ -3,80 +3,140 @@ use std::{thread, time::Duration};
 
 use crate::{Result, Error};
 
-#[cfg(all(feature = "swi", feature = "i2c"))]
-compile_error!("feature \"swi\" and feature \"i2c\" cannot be enabled at the same time");
-
-#[cfg(feature = "swi")]
 use bytes::BufMut;
-#[cfg(feature = "swi")]
-use serialport::{ClearBuffer, DataBits, SerialPort, StopBits};
-#[cfg(feature = "swi")]
-use crate::{command::EccResponse, constants::{ATCA_CMD_SIZE_MAX, WAKE_DELAY}};
+use serialport::{ClearBuffer, SerialPort};
+use crate::{constants::{WAKE_DELAY}};
 
-#[cfg(feature = "i2c")]
 use i2c_linux::{I2c, ReadFlags};
-#[cfg(feature = "i2c")]
 use std::fs::File;
 
-
-#[cfg(feature = "swi")]
-pub struct EccTransport {
-    port: String,
-}
-
-pub(crate) const RECV_RETRIES: u8 = 2;
 pub(crate) const RECV_RETRY_WAIT: Duration = Duration::from_millis(50);
+pub(crate) const RECV_RETRIES: u8 = 2;
 
-#[cfg(feature = "i2c")]
-pub struct EccTransport {
-    i2c: I2c<File>,
-    address: u16,
+pub(crate) enum TransportProtocol { 
+    I2c,
+    Swi,
+}
+pub(crate) struct EccTransport
+{
+    swi_port: Option<Box<dyn SerialPort>>, 
+    i2c_port: Option<I2c<File>>,
+    i2c_address: u16,
+    pub(crate) protocol: TransportProtocol,
 }
 
-#[cfg(feature = "i2c")]
-impl EccTransport {
+impl EccTransport
+{
     pub fn from_path(path: &str, address: u16) -> Result<Self> {
-        let mut i2c = I2c::from_path(path)?;
-        i2c.smbus_set_slave_address(address, false)?;
-        Ok(Self { i2c, address })
+        
+        if path.contains("/dev/tty"){
+            let swi_port = serialport::new(path, 230_400)
+            .data_bits(serialport::DataBits::Seven)
+            .parity(serialport::Parity::None)
+            .stop_bits(serialport::StopBits::One)
+            .timeout(Duration::from_millis(50))
+            .open().unwrap_or_else(|e| {
+                eprintln!("Failed to open serial port. Error: {}", e);
+                ::std::process::exit(1);
+            });
+            
+            Ok(Self {
+                i2c_port: None,
+                i2c_address: address,
+                swi_port: Some(swi_port),
+                protocol: TransportProtocol::Swi
+            })    
+        }
+        else {
+            let mut i2c = I2c::from_path(path)?;
+            i2c.smbus_set_slave_address(address, false)?;
+            
+            Ok(Self { 
+                swi_port: None,
+                i2c_port: Some(i2c),
+                i2c_address: address,
+                protocol: TransportProtocol::I2c
+            })
+        }
     }
 
     pub fn send_wake(&mut self) -> Result {
-        let _ = self.send_buf(0, &[0]);
-        Ok(())
+        match self.protocol {
+            TransportProtocol::I2c =>{
+                let _ = self.send_i2c_buf(0, &[0]);
+                thread::sleep(WAKE_DELAY);
+                Ok(())
+            }
+            TransportProtocol::Swi => {
+                if let Err(_err) = self.swi_port.as_mut().unwrap().set_baud_rate(115_200)
```

Comment:
Consider adding these baud rates as constant, so if ECC chip changes requiring different baud rates can be changed in one place. Also refer to datasheet or the device if it specified there.